sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/CatalystQl.scala

```diff
@@ -393,8 +393,8 @@ https://cwiki.apache.org/confluence/display/Hive/Enhanced+Aggregation%2C+Cube%2C
             overwrite)
       }
 
-      // If there are multiple INSERTS just UNION them together into on query.
-      val query = queries.reduceLeft(Union)
+      // If there are multiple INSERTS just UNION them together into one query.
+      val query = if (queries.length == 1) queries.head else Union(queries)
```

Comment:
After the merge, I am planning to introduce a new rule in `Optimizer` to detect if any child of `Union` can be pruned or if there exists only one child (e.g., after pruning), we can remove the `Union` operator from the logical plan. Not sure if this sounds good to you. Thank you!